File: testdata/bench/kotlin/BenchmarkTest00011.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Safe: parameterized query with prepared statement
suspend fun handler00011(call: ApplicationCall) {
    val id = call.parameters["id"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val stmt = conn.prepareStatement("SELECT * FROM users WHERE id = ?")
    stmt.setString(1, id)
    val rs = stmt.executeQuery()
    call.respondText(rs.toString())
}

File: testdata/bench/go/BenchmarkTest00011.go
package bench

import (
	"database/sql"
	"net/http"
)

// Parameterized query
func Handler00011(w http.ResponseWriter, r *http.Request) {
	param := r.URL.Query().Get("id")
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query("SELECT * FROM users WHERE id = ?", param)
}
